kivy_ios/tools/cpplink

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+'''
+C++ libraries are linked using environ['CXX'] compiler. This is different
+from C libraries, which are linked using environ['LDSHARED'].
+
+This script behaves like a C++ compiler or a linker, depending on
+whether its output file ends with '.so'.
+
+If for compiling, this script forwards all arguments in a call to the
+compiler specified as environ['CXX_ORIG'].
+
+If for linking, this script collects the same object and dependent
+library data.  It then generates the same files as liblink would for an
+ld target.
+'''
+
+import sys
+import subprocess
+from os import environ
+
+
+output = None
+get_next = False
+
+for arg in sys.argv:
+    if get_next:
+        output = arg
+        break
+    elif arg.startswith('-o'):
+        if arg == '-o':
+            get_next = True
+        else:
+            so_output = arg[2:]
+            break
+
+if not output.endswith('.so'):
+    result = subprocess.run([environ['CXX_ORIG'], *sys.argv[1:]])
+    sys.exit(result.returncode)
+
+
+libs = []
+objects = []
+
+i = 1
+while i < len(sys.argv):
```

Comment:
Just curious, why not using the argparse module?
https://docs.python.org/3/library/argparse.html

Reply:
Or maybe if we think argparse isn't suitable, make the code slightly more modular. That should help with readability, maintainability and testability. For instance:
```
def parse_arguments(args):
    output = None
    get_next = False
    libs = []
    objects = []
    i = 0
    while i < len(args):
        arg = args[i]
        if get_next:
            output = arg
            get_next = False
        elif arg.startswith('-o'):
            if arg == '-o':
                get_next = True
            else:
                output = arg[2:]
        elif arg.startswith("-l") or arg.startswith("-L"):
            libs.append(arg)
        elif arg in ("-r", "-pipe", "-no-cpp-precomp"):
            pass
        elif arg in ("--sysroot", "-isysroot", "-framework", "-undefined", "-macosx_version_min"):
            i += 1
        elif arg.startswith(("-I", "-m", "-f", "-O", "-g", "-D", "-arch", "-Wl", "-W", "-stdlib=")):
            pass
        elif arg.startswith("-"):
            logging.error(f"Unknown option: {arg}")
            sys.exit(1)
        elif arg.endswith('.o'):
            objects.append(arg)
        i += 1
    return output, libs, objects
```
That we could later use in a `main()` e.g.:
def main():
    # ...
    output, libs, objects = parse_arguments(sys.argv[1:])
    # ...
```
It just gives the idea, what do you think?

Reply:
Thanks for the point about cpplink being in the manifest, I have only been doing local installs.

The linking code was actually slightly modified from `tools/liblink`.  I will submit a slightly cleaned up version.

Reply:
> The linking code was actually slightly modified from tools/liblink. I will submit a slightly cleaned up version.

Oh right, that code felt familiar